src/drivers/protocol_splitter/protocol_splitter.cpp

```diff
@@ -392,6 +416,9 @@ ssize_t Mavlink2Dev::write(struct file *filp, const char *buffer, size_t buflen)
 				ret = -1;
 
 			} else {
+				_header[4] = (uint8_t) ((buflen >> 8) & 0xff);
```

Comment:
As reviewed also in https://github.com/Auterion/agent_protocol_splitter/pull/6, can we instead use the Sp2Header struct and then copy it to the buffer so this looks cleaner?